src/main/jbake/content/blog/2016/04/22/fully-non-blocking-cassandra-resultset-traversal.adoc

```diff
@@ -0,0 +1,318 @@
+= Fully non-blocking Cassandra ResultSet traversal with RxJava
+Thomas Segismont
+2016-04-22
+:icons: font
+:jbake-type: post
+:jbake-status: published
+:jbake-tags: blog, hawkular, rxjava, cassandra, asynchronous, reactive, backpressure
+
+In Hawkular Metrics, as we try to minimize resources needed to handle a massive number of clients,
+we avoid calling blocking APIs whenever possible.
+
+== Decorating Session methods with RxJava
+
+The DataStax Java driver offers methods to execute statements asynchronously. To make this work nicely with RxJava, we
+have a `Session` wrapper in our repository footnote:[https://git.io/vwGfw], exposing methods like the following:
+
+[source,java]
+.RxSession
+----
+/**
+ * Asynchronously execute a query and emit the corresponding {@link ResultSet} on the
+ * {@link Schedulers#computation()} scheduler.
+ *
+ * @param statement the CQL query to execute, of any {@link Statement} type
+ *
+ * @return an {@link Observable} emitting just one {@link ResultSet} item
+ */
+Observable<ResultSet> execute(Statement statement);
+----
+
+The implementation is quite straightforward, it simply converts the `ListenableFuture` returned by `Session#executeAsync`
+to an `Observable` with https://github.com/ReactiveX/RxJavaGuava[RxJavaGuava].
+
+[source,java]
+.RxSession implementation
+----
+@Override
+public Observable<ResultSet> execute(Statement statement) {
+    ResultSetFuture future = session.executeAsync(statement);
+    return ListenableFutureObservable.from(future, Schedulers.computation());
+}
+----
+
+IMPORTANT: Notice here that the `ResultSet` item will be emitted on the computation scheduler. This is
```

Comment:
This is displayed as red box with exclamation mark - I think this is too "aggressive". Would `INFO:` be enough?